legal-api/src/legal_api/models/filing.py

```diff
@@ -422,13 +422,14 @@ def locked(self):
 
         return False
 
-    def set_processed(self):
+    def set_processed(self, business_type):
         """Assign the completion and effective dates, unless they are already set."""
         if not self._completion_date:
             self._completion_date = datetime.utcnow()
-        if (self.effective_date is None or
-            (self.payment_completion_date
-             and self.effective_date < self.payment_completion_date)):  # pylint: disable=W0143; hybrid property
+        # In the case of an annual report or change of director filings on a COOP or BEN then the effective date can be before the payment date
+        if(self.filing_type in (Filing.FILINGS['annualReport'], Filing.FILINGS['changeOfDirectors']) and business_type in {'CP', 'BEN'}):
```

Comment:
can we move line 130 into its own function so it's easier to just use the function name to determine what is being done?  maybe name it something along lines of `can_effective_date_before_payment_completion_date`.

I don't think `Filing.FILINGS['annualReport']` works?  Maybe you are tryin gto do something like `Filing.FILINGS['annualReport'].get('name')`?